### PR TITLE
#275: add image-build-base-release command

### PR DESCRIPTION
**Pull request by Tashima42**
Closes: #275 

Adds `image-build-base-release` command. This command gets stable versions of golang, checks if there are releases in `rancher/image-build-base` and if there aren't, creates it.

⚠️⚠️ If your token has permissions to create a release in `image-build-base` it will create a new release. ⚠️⚠️
## How to test
```
export GITHUB_TOKEN={YOUR_GITHUB_TOKEN}

rke2_release image-build-base-release
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -53,4 +54,4 @@ jobs:
           push: true
           tags: rancher/ecm-distro-tools:${{ github.ref_name }}
           file: ./Dockerfile
-          context: .
+          context: .
```
Comment (briandowns) [inline .github/workflows/release.yml:57]: Can you add the new line back?

File: release/rke2/rke2.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+type goVersionRecord struct {
+	Version string `json:"version"`
+	Stable  bool   `json:"stable"`
+}
+
+type GithubRelease struct {
+	Name string `json:"name"`
+}
+
+func ImageBuildBaseRelease(ctx context.Context, githubToken string) error {
```
Comment (briandowns) [inline release/rke2/rke2.go:32]: Can you pass the github client in here? Will make this function easier to mock/test.

File: release/rke2/rke2.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	"github.com/google/go-github/v39/github"
+	ecmHTTP "github.com/rancher/ecm-distro-tools/http"
+	"github.com/rancher/ecm-distro-tools/repository"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	goDevURL           = "https://go.dev/"
+	goDevJSONPath      = "dl/?mode=json"
```
Comment (briandowns) [inline release/rke2/rke2.go:19]: It doesn't seem like the `goDevJSONPath` const needs to be a separate variable.

Comment (Tashima42) [reply release/rke2/rke2.go:19]: Yes, I was testing something and forgot to change it back

**Review by briandowns — APPROVED** [review]

**Pull request merged**